http/file_server.ts

```diff
@@ -373,7 +370,7 @@ async function serveDirIndex(
       continue;
     }
     const filePath = join(dirPath, entry.name);
-    const fileUrl = encodeURIComponent(posixJoin(dirUrl, entry.name))
+    const fileUrl = encodeURIComponent(entry.name)
```

Comment:
Not sure if this should still use some `@std/path` magic here. I considered using `posixJoin(entry.name)` but that seems redundant.